controllers/common/types.go

```diff
@@ -69,6 +70,20 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			return ctrl.Result{Requeue: true}, err
 		}
 		status.Experiment.Phase = v1alpha1.ExperimentPhaseFinished
+	} else if chaos.IsPaused() {
+		if status.Experiment.Phase == v1alpha1.ExperimentPhaseRunning {
+			r.Log.Info("Pausing")
+
+			err = r.Recover(ctx, req, chaos)
+			if err != nil {
+				r.Log.Error(err, "failed to pause chaos")
+				return ctrl.Result{Requeue: true}, err
+			}
+			status.Experiment.Phase = v1alpha1.ExperimentPhasePaused
+		} else {
+			r.Log.Error(err, "failed to pause chaos, the common chaos is not running.", "status", status.Experiment.Phase)
+			return ctrl.Result{}, errors.New("common chaos is not running")
```

Comment:
I think this message would be misleading. If we paused the chaos when it is not running, we just skip the running process and I think it's also a successful puased action.

I also suggest send an event here to indicate the chaos is paused.

Reply:
Thanks for your review, that makes more sense.
And I'm not sure if I get you about the "send an event" part. I have sent events in controller like https://github.com/pingcap/chaos-mesh/pull/370/files#diff-a52c75ede938739cb8d8e45acdfef8acR58-R62, are you referring to also send events in `{twophase,common}.Reconciler`?